### PR TITLE
pythonPackages: trezor: 0.10.2 -> 0.11.1

### PR DESCRIPTION
**Pull request by prusnak**
###### Motivation for this change

Version update

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/development/python-modules/trezor/default.nix [hunk 1]
```diff
@@ -1,19 +1,35 @@
...
+    sha256 = "1132f6a97afb0979c5018b067494bc8917b881c02d965f991270a70543b5050c";
   };
 
-  propagatedBuildInputs = [ protobuf hidapi ecdsa mnemonic requests pyblake2 click libusb1 rlp ];
+  propagatedBuildInputs = [ typing-extensions protobuf hidapi ecdsa mnemonic requests pyblake2 click construct libusb1 rlp ];
+
+  # build requires UTF-8 locale
+  preConfigure = ''
+    export LANG="en_US.UTF-8"
+    export LOCALE_ARCHIVE=${glibcLocales}/lib/locale/locale-archive
```
Comment (dotlambda) [inline pkgs/development/python-modules/trezor/default.nix]: Just put these directly in the expression as e.g. `LANG = "en_US.UTF-8";`.

Comment (prusnak) [reply pkgs/development/python-modules/trezor/default.nix]: Addressed in 9151281cb699c727713c992fe97bf4290006de89

**Comment by prusnak** [comment]
Don't merge yet, I'll amend the PR to update to upcoming 0.11.1

**Comment by prusnak** [comment]
Amended. Good to merge.

**Comment by prusnak** [comment]
@dotlambda Can you please help with the build failure? I think this is a false positive.

**Review by jb55 — APPROVED** [review]

File: pkgs/development/python-modules/trezor/default.nix [hunk 2]
```diff
@@ -1,19 +1,33 @@
...
     inherit pname version;
-    sha256 = "4dba4d5c53d3ca22884d79fb4aa68905fb8353a5da5f96c734645d8cf537138d";
+    sha256 = "6043f321d856e1b45b9df0c37810264f08d065bb56cd999f61a05fe2906e9e18";
   };
 
-  propagatedBuildInputs = [ protobuf hidapi ecdsa mnemonic requests pyblake2 click libusb1 rlp ];
+  propagatedBuildInputs = [ typing-extensions protobuf hidapi ecdsa mnemonic requests pyblake2 click construct libusb1 rlp ];
+
+  # build requires UTF-8 locale
+  LANG="en_US.UTF-8";
```
Comment (dotlambda) [inline pkgs/development/python-modules/trezor/default.nix]:
```suggestion
  LANG = "en_US.UTF-8";
```

File: pkgs/development/python-modules/trezor/default.nix [hunk 3]
```diff
@@ -1,19 +1,33 @@
...
+  LOCALE_ARCHIVE="${glibcLocales}/lib/locale/locale-archive";
```
Comment (dotlambda) [inline pkgs/development/python-modules/trezor/default.nix]: It should suffice to add `glibcLocales` to `nativeBuildInputs`.

File: pkgs/development/python-modules/trezor/default.nix [hunk 4]
```diff
@@ -1,19 +1,33 @@
...
 
   # There are no actual tests: "ImportError: No module named tests"
   doCheck = false;
```
Comment (dotlambda) [inline pkgs/development/python-modules/trezor/default.nix]: From https://github.com/trezor/python-trezor/blob/v0.11.1/tox.ini:
`python -E -m pytest --pyarg trezorlib.tests.unit_tests`

**Comment by dotlambda** [comment]
@GrahamcOfBorg eval

**Comment by prusnak** [comment]
@dotlambda addressed your comments in c220f4c, thanks for the hints!

File: pkgs/development/python-modules/trezor/default.nix (hunk 3 above)
Comment (dotlambda) [reply pkgs/development/python-modules/trezor/default.nix]: Now you should remove this line.

**Comment by prusnak** [comment]
ping @FRidh

File: pkgs/development/python-modules/trezor/default.nix [hunk 5]
```diff
@@ -1,27 +1,51 @@
...
-  propagatedBuildInputs = [ protobuf hidapi ecdsa mnemonic requests pyblake2 click libusb1 rlp ];
+  nativeBuildInputs = [ glibcLocales ];
 
-  # There are no actual tests: "ImportError: No module named tests"
-  doCheck = false;
+  propagatedBuildInputs = [ typing-extensions protobuf hidapi ecdsa mnemonic requests pyblake2 click construct libusb1 rlp ];
+
+  # build requires UTF-8 locale
+  LANG = "en_US.UTF-8";
+  LOCALE_ARCHIVE = "${glibcLocales}/lib/locale/locale-archive";
```
Comment (dotlambda) [inline pkgs/development/python-modules/trezor/default.nix]: Is `LOCaLE_ARCHIVE` still required?

Comment (prusnak) [reply pkgs/development/python-modules/trezor/default.nix]: no, fixed.

**Comment by Mic92** [comment]
```
@GrahamcOfBorg build python37Packages.keepkey python37Packages.trezor python37Packages.trezor_agent
```

File: pkgs/development/python-modules/trezor/default.nix (hunk 5 above)
Comment (dotlambda) [reply pkgs/development/python-modules/trezor/default.nix]: Any reason for this to be backported?

Comment (prusnak) [reply pkgs/development/python-modules/trezor/default.nix]: Couple of them:

1) we add a lot of features into Trezor and these are not accessible if a user has an older version of python-trezor leading to frustration
2) I assume NixPkgs Electrum maintainers don't backport security patches individually, but rather the whole package is being updated to the newest stable release. This release might already be depending on the new python-trezor package, using the new API, etc. so it makes sense to upgrade the whole package for python-trezor too.

**Pull request merged**